### PR TITLE
Better errors

### LINKED ISSUE #32: Change error handling
Currently we're handling errors by returning `{:error, "String describing error"}`. This is difficult to pattern match on, and we should move to errors that are more easy to use as a code interface (e.g. `{:error, :atom_for_error_code}` or `{:error, {:error_that_has_specific_value, "the value"}}`. 

This means we should also make a new error module that handles these errors and prints them nicely

### PR DESCRIPTION
**Pull request by Zac-Garby**
Improve validator errors by returning pattern-matchable tuples. They're all tested, too. Closes #32

### CODE REVIEW

File: lib/validator.ex [hunk 1]
```diff
@@ -70,24 +103,26 @@ defmodule UltraDark.Validator do
...
+    if tx.txtype == "COINBASE", do: :ok, else: {:error, {:not_coinbase, tx.txtype}}
   end
 
+  @spec appropriate_coinbase_output?(list, number) :: :ok | {:error, :invalid_coinbase}
   defp appropriate_coinbase_output?([coinbase | transactions], block_index) do
     total_fees = Block.total_block_fees(transactions)
     reward = Block.calculate_block_reward(block_index)
     amount = List.first(coinbase.outputs).amount
-    if D.equal?(D.add(total_fees, reward), amount), do: :ok, else: {:error, "Coinbase output is invalid"}
+    if D.equal?(D.add(total_fees, reward), amount), do: :ok, else: {:error, :invalid_coinbase}
```
Comment (alexdovzhanyn) [inline lib/validator.ex]: This should correspond directly to what makes the coinbase invalid. Saying that the coinbase is invalid isn't specific enough, as we don't know _why_ the coinbase is invalid by looking at the error message

**Comment by Zac-Garby** [comment]
@alexdovzhanyn The tweaks have been made

**Comment by alexdovzhanyn** [comment]
Alright, this is ready for merge, but we need to update the miner/wallet to account for these new errors as well

**Comment by alexdovzhanyn** [comment]
Also, there's a conflict that needs to be resolved here

**Pull request merged**